crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -712,6 +762,113 @@ impl RoomEventCacheInner {
         .await
     }
 
+    /// If the event is related to another one, it's id is added to the
+    /// relations map.
+    fn append_related_event(
+        &self,
+        cache: &mut RwLockWriteGuard<'_, AllEventsCache>,
+        event: &SyncTimelineEvent,
+    ) {
+        // Handle and cache events and relations
+        if let Ok(AnySyncTimelineEvent::MessageLike(ev)) = event.event.deserialize() {
+            if let AnySyncMessageLikeEvent::RoomRedaction(ev) = &ev {
+                match ev {
+                    SyncRoomRedactionEvent::Original(ev) => {
+                        if let Some(redacted_event_id) = &ev.content.redacts {
+                            cache
+                                .relations
+                                .entry(redacted_event_id.to_owned())
+                                .or_default()
+                                .insert(ev.event_id.to_owned());
+                        } else if let Some(redacted_event_id) = &ev.redacts {
+                            cache
+                                .relations
+                                .entry(redacted_event_id.to_owned())
+                                .or_default()
+                                .insert(ev.event_id.to_owned());
+                        }
```

Comment:
nit: let's fold these two branches:

```suggestion
                        if let Some(redacted_event_id) = ev.content.redacts.as_ref().or_else(|| ev.redacts.as_ref()) {
                            cache
                                .relations
                                .entry(redacted_event_id.to_owned())
                                .or_default()
                                .insert(ev.event_id.to_owned());
                        }
```